bpf/deserialization_to_bpf_map/deserialization_to_bpf_map.c

```diff
@@ -1727,8 +1750,22 @@ int inner_map_mng_persist()
     p->allocated_cnt = g_inner_map_mng.allocated_cnt;
     p->used_cnt = g_inner_map_mng.used_cnt;
     p->max_allocated_idx = g_inner_map_mng.max_allocated_idx;
+
+    /* Since map_fd cannot be used universally in different processes,
+    map_fd needs to be converted into map_id and stored, and then
+    converted into map_fd in the corresponding process when the
+    configuration is restored.
+    When storing map_fd into outer_map, the kernel update interface will
+    perform special processing on maps of types BPF_MAP_TYPE_ARRAY_OF_MAPS
+    and BPF_MAP_TYPE_HASH_OF_MAPS. The input parameter is map_fd, but the
+    kernel will convert it into a bpf_map pointer for storage, so it will
+    not be affected.*/
     for (i = 0; i <= g_inner_map_mng.max_allocated_idx; i++) {
-        p->inner_map_stat[i].map_fd = g_inner_map_mng.inner_maps[i].map_fd;
+        int map_fd = bpf_get_map_id_by_fd(g_inner_map_mng.inner_maps[i].map_fd);
```

Comment:
Retaining the restarted configuration may fail. If the failure occurs, the configuration persistence process is abandoned and all saved configurations will be deleted.